### PR TITLE
Mc 1.17 refactor reorganize

### PR DESCRIPTION
**Pull request by Niko-sk2x**
Resubmitting from https://github.com/OpenCubicChunks/CubicChunks/pull/737

Review commit by commit.

### CODE REVIEW

**Comment by Niko-sk2x** [comment]
This is ready for review

**Comment by NotStirred** [comment]
Perhaps add a`.git-blame-ignore-revs` specifying these commits

File: src/main/java/io/github/opencubicchunks/cubicchunks/client/multiplayer/ClientCubeCacheStorage.java [hunk 1]
```diff
@@ -39,35 +38,35 @@ public int getIndex(int x, int y, int z) {
             + Math.floorMod(x, this.horizontalSideLength);
     }
 
-    public void replace(int cubeIdx, @Nullable BigCube chunkIn) {
-        BigCube cube = this.cubes.getAndSet(cubeIdx, chunkIn);
-        if (cube != null) {
+    public void replace(int cubeIndex, @Nullable LevelCube replaceWith) {
+        LevelCube oldCUbe = this.cubes.getAndSet(cubeIndex, replaceWith);
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/client/multiplayer/ClientCubeCacheStorage.java]: typo

File: src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/blockentity/MixinConduitBlockEntity.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@Mixin(ConduitBlockEntity.class)
+public class MixinConduitBlockEntity {
+    /**
+     * @author NotStirred
+     * @reason Conduits now only affect players within +256 of the block. This prevents near infinite cube loading in the conduit's column
+     */
+    // TODO: NotStirred merge this with BeaconTileEntity to reduce duplicated code
+    // TODO: redirect getHeight()
+    @Redirect(method = "applyEffects", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/phys/AABB;expandTowards(DDD)Lnet/minecraft/world/phys/AABB;"))
+    private static AABB on$expand(AABB aabb, double x, double y, double z, Level level, BlockPos pos, List<BlockPos> activatingBlocks) {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/blockentity/MixinConduitBlockEntity.java]: rename to match `MixinBeaconBlockEntity#limitBeaconEffectYRange`

**Comment by NotStirred** [comment]
rename `MixinChunkMap#on_func_219244_a_StatusChange`

File: src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/chunk/MixinLevelChunk.java [hunk 3]
```diff
@@ -278,8 +277,6 @@ private void setIsModifiedFromSetBlockState_Field(LevelChunk chunk, boolean isMo
             this.unsaved = isModifiedIn;
             return;
         }
-
-
 //        if (isColumn) {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/chunk/MixinLevelChunk.java]: Remove 1.12 logic, replaced by isCubic check

File: src/main/java/io/github/opencubicchunks/cubicchunks/server/level/CubeMap.java [hunk 4]
```diff
@@ -8,54 +8,66 @@
...
-    CompletableFuture<Either<BigCube, ChunkHolder.ChunkLoadingFailure>> getCubeEntityTickingRangeFuture(CubePos pos);
+    // prepareEntityTickingChunk
+    CompletableFuture<Either<LevelCube, ChunkHolder.ChunkLoadingFailure>> prepareEntityTickingCube(CubePos pos);
 
+    // getChunks
     Iterable<ChunkHolder> getCubes();
 
-    // func_219215_b, checkerboardDistance
+    // checkerboardDistance
     // replacement of func_219215_b, checkerboardDistance, checks iew distance instead of returning distance
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/server/level/CubeMap.java]: remove srg names

**Review by NotStirred — CHANGES_REQUESTED** [review]
Some minor things to be changed:

`ImposterProtoCube` commented out code at line @ 243

all remaining srg `field_xxxxxx_x` names
```
MixinChunkMap.java:155
MixinChunkMap.java:157
```

all remaining srg `p_` names

```
PlayerTicketTrackerFactoryAccess.java:11
FixedPlayerDistanceChunkTrackerFactoryAccess.java:11
MixinClientLevel.java:48
MixinClientLevel.java:49
MixinChunkHolder.java:103
MixinChunkMap.java:347
MixinChunkMap.java:348
MixinChunkMap.java:540
MixinChunkMap.java:849
MixinChunkMap.java:850
MixinChunkMap.java:874
MixinDistanceManager.java:71
MixinLayerLightSectionStorage.java:47
CubeMap.java:59
CubeMap.java:60
RegionCubeIO.java:124
RegionCubeIO.java:315
RegionCubeIO.java:316
CubePos.java:98
CubePos.java:99
CubePos.java:100
CubePos.java:101
ImposterProtoCube.java:193
ImposterProtoCube.java:194
ImposterProtoCube.java:197
ImposterProtoCube.java:236
ImposterProtoCube.java:244
ImposterProtoCube.java:245
ImposterProtoCube.java:250
ImposterProtoCube.java:251
```

all remaining srg `func_` names
```
MixinChunkMap.java:513
MixinChunkMap.java:518
MixinChunkMap.java:519
MixinChunkMap.java:1209
MixinChunkMap.java:1250
CubeMap.java:71
CubicPlayerTicketTracker.java:112
```

**Review by NotStirred — APPROVED** [review]

**Pull request merged**